### PR TITLE
Create group_partitions option & allow direct control over train/test split

### LINKED ISSUE #182: Provide option for users to define train/test split and CV splits
This provides more flexibility, e.g. if users want to do leave-one-out cross-validation, or split into certain groups.

### PR DESCRIPTION
**Pull request by kelly-sovacool**
## Issues
- Resolves #182. 

## Change(s) made
- Create `group_partitions` option.
  - Currently, it influences the train/test set but _not_ the CV splits.
- Allow users to supply training set indices directly instead of a `training_frac`.

## Checklist

(~Strikethrough~ any points that are not applicable.)

- [x] Write unit tests for any new functionality or bug fixes.
- [x] Update roxygen comments & vignettes if there are any API changes.
- [x] Update `NEWS.md` if this includes any user-facing changes. 
- [x] The check workflow succeeds on your most recent commit. **This is always required before the PR can be merged.**

### CODE REVIEW

File: R/partition.R [hunk 1]
```diff
@@ -37,42 +42,87 @@ get_partition_indices <- function(outcomes, training_frac = 0.8, groups = NULL)
...
+      if (num_needed > 0) {
+        more_train_samples <- indices[sample(remaining, size = num_needed)]
+        train_set <- c(train_set, more_train_samples)
+      }
+    }
   }
-  message(paste0("Fraction of data in the training set: ", frac_in_train, "."))
-  # get train groups indices
+
+  frac_in_train <- length(train_set) / length(indices)
```
Comment (courtneyarmour) [inline R/partition.R:117]: should we include a warning for certain scenarios that might be unwanted (i.e. training set smaller than test set)?

File: R/checks.R [hunk 2]
```diff
@@ -124,6 +123,48 @@ check_training_frac <- function(frac) {
...
+      sep = "\n"
+    )
+  }
+  if (max(training_inds) > nrow(dataset)) {
+    stop_msg <- paste0(stop_msg,
+      "The training indices vector contains a value that is too large for the number of rows in the dataset.",
+      sep = "\n"
+    )
+  }
+  if (length(training_inds) > nrow(dataset)) {
```
Comment (courtneyarmour) [inline R/checks.R:157]: should this be >= because the whole dataset cant be in the training set right? or maybe an addtional error for length(training_inds) == nrow(dataset)

File: R/partition.R (hunk 1 above)
Comment (kelly-sovacool) [reply R/partition.R:117]: Good suggestion, added here: 6d13ec0

File: R/checks.R (hunk 2 above)
Comment (kelly-sovacool) [reply R/checks.R]: Fixed here: 0536eb6

**Review by courtneyarmour — APPROVED** [review]

File: vignettes/introduction.Rmd [hunk 3]
```diff
@@ -235,6 +235,29 @@ This is okay for testing things out and may even be necessary for smaller datase
...
 we think the defaults are a good starting point [@topcuoglu_framework_2020].
 
+### Custom training indices
+
+When `training_frac` is a fraction between 0 and 1, a random sample of
+observations in the dataset are chosen for the training partition to satisfy the
+`training_frac`. However, in some cases you might wish to control exactly which
+observations are in the training partition. You can instead assign
+`training_frac` a vector of indices that correspond to which rows of the dataset
+should go in the training fraction of the train/test split.
```
Comment (zenalapp) [inline vignettes/introduction.Rmd:245]: And everything else goes into the testing fraction?

Comment (kelly-sovacool) [reply vignettes/introduction.Rmd:245]: Yep!

Comment (zenalapp) [reply vignettes/introduction.Rmd:245]: Cool, just wanted to make sure

Comment (kelly-sovacool) [reply vignettes/introduction.Rmd:245]: We could add that:
```suggestion
should go in the training fraction of the train/test split 
(all remaining sequences will go in the testing fraction).
```

Comment (zenalapp) [reply vignettes/introduction.Rmd:245]: Perfect

Comment (kelly-sovacool) [reply vignettes/introduction.Rmd:245]: Adding this here: https://github.com/SchlossLab/mikropml/pull/282
